### PR TITLE
expand map buttons more for long texts

### LINKED ISSUE #4709: Map buttons do not extend enough to show full text in some languages
https://user-images.githubusercontent.com/2395284/129546954-ac1be045-ffbc-4545-b58b-47be92617533.mp4

### PR DESCRIPTION
**Pull request by Zayankovsky**
### Description
Fixes #4709. 

### Screenshots or Gifs
https://user-images.githubusercontent.com/2395284/129546379-4ef9bd5e-1e7d-453e-9c49-669f1f8f86fd.mp4

### CODE REVIEW

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/view/MapboxRecenterButton.kt [hunk 1]
```diff
@@ -145,6 +146,14 @@ class MapboxRecenterButton : ConstraintLayout {
         )?.let { binding.recenterText.setTextColor(it) }
     }
 
+    private fun getExtendToWidth(text: String): Int {
```
Comment (alexander-kulikovskii) [inline libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/view/MapboxRecenterButton.kt:149]: is it possible to move this function to the shared code?

maybe as extension for textView

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/view/MapboxRecenterButton.kt [hunk 2]
```diff
@@ -145,6 +146,14 @@ class MapboxRecenterButton : ConstraintLayout {
...
+        val measuredWidth = binding.recenterText.paint.measureText(text.uppercase())
+        val paddings = binding.recenterText.paddingStart + binding.recenterText.paddingEnd
```
Comment (alexander-kulikovskii) [inline libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/view/MapboxRecenterButton.kt:151]: nit:
```suggestion
        val horizontalPaddings = binding.recenterText.paddingStart + binding.recenterText.paddingEnd
```

File: libnavui-util/src/main/java/com/mapbox/navigation/ui/utils/internal/extensions/TextViewEx.kt [hunk 3]
```diff
@@ -18,3 +18,8 @@ fun TextView.getAsBitmap(): Bitmap {
     draw(canvas)
     return bitmap
 }
+
+fun TextView.measureText(text: String): Float {
```
Comment (alexander-kulikovskii) [inline libnavui-util/src/main/java/com/mapbox/navigation/ui/utils/internal/extensions/TextViewEx.kt:22]:
```suggestion
fun TextView.measureTextWidth(text: String): Float {
```

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/view/MapboxRouteOverviewButton.kt [hunk 4]
```diff
@@ -88,13 +89,14 @@ class MapboxRouteOverviewButton : ConstraintLayout {
     fun showTextAndExtend(duration: Long) {
         if (!isAnimationRunning) {
             isAnimationRunning = true
-            val extendToWidth = EXTEND_TO_WIDTH * context.resources.displayMetrics.density
+            val text = context.getString(R.string.mapbox_route_overview)
+            val extendToWidth = (binding.routeOverviewText.measureText(text) + textWidth)
```
Comment (alexander-kulikovskii) [inline libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/view/MapboxRouteOverviewButton.kt:93]: what's the difference between `textWidth` and `...measureText(text)`?

**Review by alexander-kulikovskii — APPROVED** [review]

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/view/MapboxRouteOverviewButton.kt (hunk 4 above)
Comment (Zayankovsky) [reply libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/view/MapboxRouteOverviewButton.kt]: `textWidth` is actually width of the text view when it is shrunk and no text is shown. 
I renamed the property to reflect that.

**Pull request merged**